### PR TITLE
[#3949] add tests for res.metadata.update()

### LINKED ISSUE #3949: resource.metadata.update doesn't work properly
Max and I tried to use this function to test "export to dataverse" functionality and found that: 
1. The documentation on how to use it is incorrect. 
2. Updates of selected fields -- notably creator -- silently fail to do anything.

### PR DESCRIPTION
**Pull request by devincowan**
Resolves #3949 

### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

- [ ] Automated Testing

- [ ] Sufficient User and Developer Documentation

- [ ] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. Tests don't break. Seems the res.metadata.update() function is behaving as we expect

### CODE REVIEW

**Comment by hydrocheck** [comment]
| METRIC | VALUE |
|:-----:|:-----:|

[https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4845](https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4845)

File: hs_core/tests/api/native/test_update_metadata.py [hunk 1]
```diff
@@ -21,6 +21,173 @@ def setUp(self):
...
+        metadata.append({'subject': {'value': 'sub-3'}})
+        self.res.metadata.update(metadata, self.user)
+        # there should be 2 subject elements for this resource
+        self.assertEqual(self.res.metadata.subjects.all().count(), 2, msg="Number of subject elements found not be 2.")
+        self.assertIn('sub-3', [sub.value for sub in self.res.metadata.subjects.all()],
+                      msg="Subject element with value of %s does not exist." % 'sub-3')
+        self.assertIn('sub-4', [sub.value for sub in self.res.metadata.subjects.all()],
+                      msg="Subject element with value of %s does not exist." % 'sub-4')
+
     def test_update_science_metadata(self):
```
Comment (pkdash) [inline hs_core/tests/api/native/test_update_metadata.py:191]: Do we still need this test?

File: hs_core/tests/api/native/test_update_metadata.py [hunk 2]
```diff
@@ -21,6 +21,173 @@ def setUp(self):
 
         self.res = hydroshare.create_resource('GenericResource', self.user, 'Test Resource')
 
+    def test_resource_metadata_update(self):
+        # add  metadata elements
+        metadata_dict = [
```
Comment (pkdash) [inline hs_core/tests/api/native/test_update_metadata.py:26]: `fundingagency` is missing

File: hs_core/tests/api/native/test_update_metadata.py (hunk 1 above)
Comment (devincowan) [reply hs_core/tests/api/native/test_update_metadata.py:191]: good point, this is duplicative with the test added in this PR. I guess that I had thought we might find an issue testing each metadata type with an individual update (which in hindsight was silly). Anyway, I'm just going to close this PR. Thanks Pabitra

**Pull request closed without merging**